app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -3,14 +3,17 @@ module RemoteExecution
     class RunHostsJob < Actions::ActionWithSubPlans
 
       middleware.use Actions::Middleware::BindJobInvocation
+      middleware.use Actions::Middleware::RecurringLogic
 
-      def delay(delay_options, job_invocation)
+      def delay(delay_options, job_invocation, locked = false)
+        task.add_missing_task_groups(job_invocation.task_group)
```

Comment:
Currently, the iterations after the first one don't have their own job invocation instance, so that the events are not shown in job invocations index, so we then can't track status from the job perspective then. We should check here, and in case the job has already a task, we should clone the invocation, rather than use the current one.

Also, since we're mapping the job invocation to 0-1 tasks, we should rename 'last_task_id', to just 'task_id'.